src/zenml/new/pipelines/pipeline.py

```diff
@@ -642,6 +645,14 @@ def _run(
             constants.SHOULD_PREVENT_PIPELINE_EXECUTION = True
             try:
                 stack.deploy_pipeline(deployment=deployment_model)
+            except Exception as e:
+                runs = Client().list_pipeline_runs(
+                    deployment_id=deployment_model.id,
+                    sort_by="desc:start_time",
+                    size=1,
+                )
+                self.delete_running_versions_without_recovery(runs.items[0])
+                raise e
```

Comment:
Multiple runs can be linked to the same deployment, so this might not be safe. E.g., if I have a pipeline that I run on schedule every 1h, but each run takes >1h, then it could happen that this call queries the wrong pipeline run

Reply:
Ah, if you only need the configs, then here's an easy fix: The deployment already has both the pipeline config and all step configs in it, so you can just pass that into `delete_running_versions_without_recovery()` directly

Reply:
Good catch, I will include this into https://zenml.atlassian.net/browse/OSS-2463. I don't have runs linked yet.

Reply:
Would suggest to change it in this PR directly so we don't forget - after #1835 gets merged the runs and steps themselves won't even have configs anymore so your current will be broken in general

Reply:
Ah, this one was only about runs, got it now. I read it a bit differently: what if I have a scheduled pipeline and it creates a new version of the model? Runtime is 2h and the schedule is every 1h - there will be a clash and mess in versions, but for this, I need to link pipeline runs first.

I prepared a fix for passing deployment object into `get_new_version_requests` 👍🏼